api/XDS_PROTOCOL.md

```diff
@@ -286,50 +286,50 @@ admin:
 
 ```
 
-### Incremental xDS
+### Delta xDS
 
-Incremental xDS is a separate xDS endpoint available for ADS, CDS and RDS that
+Delta xDS is a separate xDS endpoint available for ADS, CDS and RDS that
```

Comment:
This endpoint does both on-demand and delta. So, I think we want to call this incremental.

Reply:
Isn't this section just talking about the xDS communication itself, i.e. what api_type you specify and therefore whether you're using the code this PR adds? In that case, this really is just delta. On-demand is possible even without delta (albeit horribly inefficiently), right? And on-demand is all just about giving Envoy's actual logic the ability to call into xDS-land as part of the datapath.

Reply:
On-demand isn't possible with non-delta using regular xDS. Take something like CDS as an example. There is no way, using regular CDS, for Envoy to ask for additional clusters, since it expects state-of-the-world CDS updates, that's baked into the protocol. There are ways to build on-demand using custom filters, sure, but then you need to make RPCs from those filters using some custom protocol and manipulate the `ClusterManager` directly.

Ultimately, when it comes to the wire format, it's basically xDS next generation, and it's there to support both delta updates and on-demand, plus some other goodies like partial rejects perhaps, finer grained resource versioning..

So, I think it would be better to make the protos and this section Incremental.

Reply:
I thought the existing protocol has Envoy specifying all of the names it's interested in with every Request message (with no names meaning "all of them"). And, I thought on-demand was expected to be implemented within the filter chain even with this incremental/delta xDS work.

Reply:
>I thought the existing protocol has Envoy specifying all of the names it's interested in with every > Request message (with no names meaning "all of them"). 

Yep.

> And, I thought on-demand was expected to be implemented within the filter chain even with this > incremental/delta xDS work.

On-demand is expected to be implemented on a per-xDS basis, and for clusters (for example) will be handle by `ClusterManager` and `CdsApiImpl`, so there is no filter chain aspect to this.

Reply:
Got it; done.